django/utils/text.py

```diff
@@ -314,8 +316,27 @@ def phone2numeric(phone):
     return "".join(char2number.get(c, c) for c in phone.lower())
 
 
-def compress_string(s):
-    return gzip_compress(s, compresslevel=6, mtime=0)
+def _get_random_fname(max_random_bytes):
+    return "a" * secrets.randbelow(max_random_bytes)
+
+
+def compress_string(s, max_random_bytes=0):
+    compressed_data = gzip_compress(s, compresslevel=6, mtime=0)
+
+    if max_random_bytes == 0:
+        return compressed_data
+
+    header = bytearray(compressed_data[:10])
+    header[3] = gzip.FNAME
+
+    return b"".join(
+        [
+            header,
+            _get_random_fname(max_random_bytes).encode("ascii"),
+            b"\x00",
+            compressed_data[10:],
```

Comment:
```suggestion
            memoryview(compressed_data)[10:],
```

Using `memoryview` seems to prevent creating a copy of bytes for the slice. At least in the test that I used:
```sh
$ /usr/bin/time -f "%M" python -c 'b = b"0" * 10**6; b"".join([b"123", b[10:]])'            
11880
$ /usr/bin/time -f "%M" python -c 'b = b"0" * 10**6; b"".join([b"123", memoryview(b)[10:]])'
10820
```